algorithms/linfa-clustering/src/dbscan/algorithm.rs

```diff
@@ -71,27 +74,60 @@ use linfa::{DatasetBase, Float};
 pub struct Dbscan;
 
 impl Dbscan {
-    pub fn params<F: Float>(min_points: usize) -> DbscanHyperParamsBuilder<F> {
-        DbscanHyperParams::new(min_points)
+    /// Configures the hyperparameters with the minimum number of points required to form a cluster
+    ///
+    /// Defaults are provided if the optional parameters are not specified:
+    /// * `tolerance = 1e-4`
+    /// * `dist_fn = L2Dist` (Euclidean distance)
+    /// * `nn_impl = KdTree`
+    pub fn params<F: Float>(
+        min_points: usize,
+    ) -> DbscanHyperParamsBuilder<F, L2Dist, CommonNearestNeighbour> {
+        Self::params_with_custom_fields(min_points, L2Dist, CommonNearestNeighbour::KdTree)
+    }
+
+    /// Configures the hyperparameters with the minimum number of points, a custom distance metric,
+    /// and a custom nearest neighbour algorithm
+    pub fn params_with_custom_fields<F: Float, D: Distance<F>, N: NearestNeighbour>(
```

Comment:
the postfix `custom_fields` sounds a bit too vague, why no `params_with_metric_nn` or just `params_with`